File: Algorithms/Paradigms/Greedy Algorithms/Activity Selection Problem/activity-selection-problem.py
"""
    Activity Selection Problem
"""

# n --> Total number of activities
# s[]--> An array that contains start time of all activities
# f[] --> An array that conatins finish time of all activities


def print_max_activities(s , f ):
    print("The following activities are selected")

    # The first activity is always selected
    i = 0
    print(i)

    for j in range(len(f)):

        # If this activity has start time greater than
        # or equal to the finish time of previously
        # selected activity, then select it
        if s[j] >= f[i]:
            print(j)
            i = j


s = [1, 3, 0, 5, 8, 5]
f = [2, 4, 6, 7, 9, 9]
print_max_activities(s, f)

File: Algorithms/Paradigms/Greedy Algorithms/Activity Selection Problem/activity-selection-problem.cpp
/*
 * Activity Selection Problem
 *
 */

#include <iostream>

using namespace std;

// n --> Total number of activities
// s[]--> An array that contains start time of all activities
// f[] --> An array that conatins finish time of all activities
void print_max_activities(int s[], int f[], int n) {
    int i, j;
    i = 0;
    cout<<i<<endl;

    for (j = 1; j < n; j++) {
        if (s[j] >= f[i]) {
            cout<<j<<endl;
            i = j;
        }
    }
}

int main() {
    int s[] =  {1, 2, 6, 7, 8, 5};
    int f[] =  {2, 6, 6, 9, 9, 9};
    int n = sizeof(s)/sizeof(s[0]);
    print_max_activities(s, f, n);
    return 0;
}
